File: src/main/java/com/github/biconou/audioplayer/legacy/AudioSystemUtils.java
package com.github.biconou.audioplayer.legacy;

/*-
 * #%L
 * newaudioplayer
 * %%
 * Copyright (C) 2016 - 2017 Rémi Cocula
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * #L%
 */

import javax.sound.sampled.*;
import java.util.Arrays;
import java.util.function.Predicate;

/**
 * Created by remi on 19/03/17.
 */
@Deprecated
public class AudioSystemUtils {



    public static Mixer.Info[] listAllMixers() {
        return AudioSystem.getMixerInfo();
    }

    public static Mixer findMixerByName(String mixerName) {

        Mixer.Info[] allMixers = AudioSystemUtils.listAllMixers();

        Predicate<Mixer.Info> p = mixer -> mixer.getName().equals(mixerName);
        Mixer.Info found = Arrays.stream(allMixers).filter(p).findFirst().get();
        if (found != null) {
            return AudioSystem.getMixer(found);
        } else {
            return null;
        }
    }



}

File: src/main/kotlin/com/github/biconou/kaudio/audio/system/AudioSystemUtils.kt
package com.github.biconou.kaudio.audio.system

import com.github.biconou.kaudio.audio.format.convertToPCMAudioFormat
import java.io.File
import java.io.FileNotFoundException
import javax.sound.sampled.AudioInputStream
import javax.sound.sampled.AudioSystem
import javax.sound.sampled.Mixer
import javax.sound.sampled.UnsupportedAudioFileException


fun listAllMixers(): Array<Mixer.Info> {
    return AudioSystem.getMixerInfo()
}

fun findMixerByName(mixerName: String): Mixer? {
    val found = listAllMixers().filter { m -> m.name == mixerName }
    if (found.isEmpty()) {
        return null
    } else {
        return AudioSystem.getMixer(found.first())
    }
}

@Throws(FileNotFoundException::class)
fun getPCMAudioInputStream(file: File): AudioInputStream {

    val sourceAudioInputStream: AudioInputStream? = AudioSystem.getAudioInputStream(file)

    return sourceAudioInputStream?.run {
        AudioSystem.getAudioInputStream(convertToPCMAudioFormat(format), this)
    } ?: throw UnsupportedAudioFileException()
}



